### PR TITLE
Add raml and schemas for groups and rules.  Update raml for jobs and …

### PR DESCRIPTION
**Pull request by coltonlw**
Adds RAML and schemas for jobs, gears, groups.  Complete rules output raml and schemas.

Changes groups handler to use separate schemas for new and update, which $ref the group input schema definition.  Tests added for invalid input to groups endpoint to verify that change.

### Review Checklist

- Tests were added to cover all code changes
- Documentation was added / updated
- Code and tests follow standards in CONTRIBUTING.md

…gears

### CODE REVIEW

**Comment by gsfr** [comment]
Great. Thanks, @coltonlw. I think @nagem should do the review for this.

**Comment by coltonlw** [comment]
Sounds good - no rush Megan hopefully the github notifications don't disturb your weekend

To make it clear how the tests for /groups 400 (invalid input) get enabled, they're turned on "automatically" by abao when a status 400 response for that endpoint is added to the raml.  In this case I'm using a [RAML trait](http://raml.org/developers/raml-200-tutorial#traits) called "validates-json-body" which adds a 400 response to any endpoint with that trait.  The trait definition is in api.raml

File: raml/schemas/definitions/group-output.json [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+          "_id": {
+            "type": "string"
+          },
+          "site": {
+            "type": "string"
+          }
+        },
+        "required": [
+          "access",
+          "_id"
```
Comment (nagem) [inline raml/schemas/definitions/group-output.json]: `roles` on groups and `permissions` on projects/sessions/acquisitions will all follow the same schema, they just have different field names (`roles` vs `permissions`). We'll probably want to add a schema snippet for that.

**Comment by coltonlw** [comment]
@nagem got that taken care of

**Comment by coltonlw** [comment]
Blocking #404 , we need this first because it corrects job output schema.  Otherwise I have changes ready to resolve #404 and enable jobs test plus a postman workflow test that includes uploading a file to an acquisition.

**Comment by nagem** [comment]
This LGTM but there is a lot to go over. Even if we did miss something, it's safe to merge. I'd say go ahead.

**Pull request merged**